src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/Designer/ILaunchProfileEditorClientSession.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug.Designer
+{
+    /// <summary>
+    /// Proffered by the client, for notifications from the server.
+    /// </summary>
+    internal interface ILaunchProfileEditorClientSession
+    {
+        /// <summary>
+        /// Provides an editor client with the initial snapshot of properties, including all metadata and values.
+        /// </summary>
+        Task ReceivePropertiesAsync(
+            ImmutableArray<Property> properties);
+
+        /// <summary>
+        /// Notifies an editor client of a change to the values of one or more properties.
+        /// </summary>
+        /// <returns>
+        /// A Task that completes when the client has received and scheduled the property update.
+        /// </returns>
+        Task ReceivePropertyUpdatesAsync(
+            Guid? correlationId,
+            ImmutableArray<PropertyUpdate> propertyUpdates);
```

Comment:
Now that I've posted this, I think this is insufficient as an update mechanism. If the user adds a new profile, then we need to communicate a whole set of additional properties, metadata _and_ values, back to the client.

Reply:
This is largely the approach I ended up with.